### PR TITLE
pkg/daemon/update.go: check wait err and wrap it

### PR DESCRIPTION
**Pull request by runcom**
we missed checking it https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/update.go#L92

also, refactored the condition function a bit to be more future proof (i.e. someone comes and add a `return false, err`)

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

**Comment by runcom** [comment]
aws :heart:

**Comment by ashcrow** [comment]
> aws heart

Yeah. Looks like resources.

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -89,23 +89,25 @@ func (dn *Daemon) updateOSAndReboot(newConfig *mcfgv1.MachineConfig) error {
...
 				IgnoreDaemonsets:   true,
 			})
-			if err != nil {
-				glog.Infof("Draining failed with: %v; retrying...", err)
-				lastErr = err
-				return false, nil
+			if err == nil {
+				return true, nil
+			}
+			lastErr = fmt.Errorf("draining failed with: %v", err)
```
Comment (jlebon) [inline pkg/daemon/update.go]: How about just doing `glog.Infof()` for the error message here and leaving `lastErr = err`? Then if we wrap it later, we can still get back the original error that happened. (Anyway, "failed to drain node (5 tries): draining failed with: ..." is kinda repetitive, no? :))

Comment (runcom) [reply pkg/daemon/update.go]: :+1:

Comment (runcom) [reply pkg/daemon/update.go]: fixed, thanks!

**Pull request merged**